iceoryx_posh/include/iceoryx_posh/iceoryx_posh_types.inl

```diff
@@ -49,6 +50,16 @@ log::LogStream& operator<<(log::LogStream& stream, ConnectionState value) noexce
     return stream;
 }
 
+ResourcePrefix_t iceoryxResourcePrefix(uint16_t uniqueRouDiID, iox::string<1> customizer)
```

Comment:
So the customizer is a prefix suffix. What do you think about an enum class called `IceoryxResourceTypes` and then deduce the prefix suffix from it? The user then just generates an iceoryx resource prefix for a specific type and everything is nicely readable.

Reply:
We just need to somehow distinguish between internal/static and external/dynamic resources. I think even the RouDi IPC channel should have a name which cannot be set by a runtime. I forgot about that. So how about
```cpp
enum class ResourceType {
    ICEORYX_DEFINED,
    USER_DEFINED,
};
```